### DISCUSSION TITLE
How to get the elements after rerender

### DISCUSSION THREAD

**Discussion by 19Qingfeng** [testing-library/testing-library-docs#1290] [Q&A]
I have reviewed the documentation, but still haven't found how to obtain the re rendered elements, eg: 

```js
    const ref = createRef<TabInstance>();
    const { rerender,getByRole } = render(<TabComponent headerRef={ref} />);

    const input = getByRole('textbox')

    expect(ref.current).toBeDefined();

    rerender(<TabComponent headerRef={ref} value="3" />)
    
    // May I ask how to obtain the rendered  `const input = getByRole('textbox')` element

```

I'm not sure how I should obtain it, I haven't found any valid information on the document

**Comment by MatanBobi** [comment]
Hi @19Qingfeng.
I'm not sure I'm following the purpose of this test.
Using `ref` inside a test isn't a best practice since when we're testing we should act as if we're the user using our app.
If you want to test this component, I'd assert on what the user is actually seeing, if the `value` 3 is presented, I'd assert that.
